### ISSUE TITLE
[11.0] Delivery quantity doesn't update on barcode scanning after scanning unreserved product

### ISSUE THREAD

**Issue by cpintofonseca** [odoo/odoo#24922]
DImpacted versions:
11.0

Steps to reproduce:
On Odoo 11.0 Enterprise runbot (to avoid listing the installed modules) create a new Sale Order with the product that has the following barcode: 054398267125. After that, confirm the Sale Order and navigate to the related delivery. Please make sure the product is NOT reserved at this point. Click in "Edit" and try to scan the barcode of the unreserved product, it will not work as expected but them something happens. Click in "Check Availability" to reserve the product, click in "Edit" and try to scan again.
May be related with this issue: https://github.com/odoo/odoo/issues/24872

Current behavior:
If you scan a barcode of a product that is NOT reserved then reserve the product and scan again, the "Done" quantity doesn't update (even if you click in the "Save" button).

Expected behavior:
If you scan a barcode of a product that is NOT reserved then reserve the product and scan again, the "Done" quantity should be updated on the second attempt since the product is reserved.

Video/Screenshot link (optional):
https://youtu.be/YkGKZCN4EpE

**Comment by pedrobaeza** [comment]
But why are you clicking on edit? I think the barcode scanner works without being in edit mode. Have you tried?

**Comment by amoyaux** [comment]
Could you please try again by using the barcode app or check 'Show details operation' on your delivery order picking type.

**Comment by cpintofonseca** [comment]
@pedrobaeza it only works in edit mode.

@amoyaux this works fine using the barcode app or "Show Detailed Operations" on the picking type.

I know this new feature was added part of the inventory refactor on 11.0, but still in my opinion, we should be able to use the barcode scanner without the barcode app or "Show Detailed Operations" like we used to on 10.0, unless this is not intended anymore and maybe there's a good reason for that.
